### DISCUSSION TITLE
InputPicker / SelectPicker event.target.id empty

### DISCUSSION THREAD

**Discussion by mpijorr** [rsuite/rsuite#2594] [General]
I use the property id="xxx" on the InputPicker or SelectPicker, the event callback function doesn't return the event.target.id. They are always empty.
Is this the default behavior?

**Comment by SevenOutman** [comment]
SelectPicker consists of multiple HTML elements and it's not very clear which extra props should be passed to which element. Which event callback are you using and how would you like to make use of the `event.target.id`?

**Comment by mpijorr** [comment]
Thanks for the response. I have multiple Picker components and would like to use just one call back function (onSelect) to get the value responding to the Picker Id.

**Reply by SevenOutman** [reply]
That might not work though. `onSelect` is not a real event callback that's triggered by a "select" event on the picker. Instead it's triggered by a click event on an option, thus the `event.target` wouldn't be the picker itself but the option being selected.

I'd suggest you use separate callbacks for each picker, or use `<Form>` component if you want to compose a form.
